index.js

```diff
@@ -532,10 +583,16 @@ class KeyringController extends EventEmitter {
       }),
     );
     const encryptedString = await this.encryptor.encrypt(
-      this.password,
+      this.encryptedKey,
       serializedKeyrings,
     );
-    this.store.updateState({ vault: encryptedString });
+
+    const newVault = [encryptedString, VAULT_SEPARATOR, salt].join('');
```

Comment:
(could) assert both values are not undefined 
If we introduce a bug somewhere in the future it'd be better to throw an error and leave the password encrypted vault as is than to save something like `VAULT:::undefined` accidentally

Reply:
Good idea!